services/brig/src/Brig/API/User.hs

```diff
@@ -213,6 +215,73 @@ verifyUniquenessAndCheckBlacklist uk = do
       unless av $
         throwE IdentityErrorUserKeyExists
 
+createUserSpar :: NewUserSpar -> ExceptT CreateUserSparError (AppT r) CreateUserResult
+createUserSpar new = do
+  let handle' = newUserSparHandle new
+      new' = newUserFromSpar new -- TODO: make it obsolete once we add rich info?
+      ident = newUserSparSSOId new
+      tid = newUserSparTeamId new
+
+  -- Create account
+  account <- lift $ do
+    (account, pw) <- wrapClient $ newAccount new' Nothing (Just tid) handle'
+
+    let uid = userId (accountUser account)
+
+    -- TODO: make this transactional?
+    wrapClient $ Data.insertAccount account Nothing pw False
+    case unRichInfo <$> newUserSparRichInfo new of
+      Just richInfo -> wrapClient $ Data.updateRichInfo uid richInfo
+      Nothing -> pure () -- Nothing to do
+    wrapHttp $ Intra.createSelfConv uid
+    wrapHttpClient $ Intra.onUserEvent uid Nothing (UserCreated (accountUser account))
+
+    pure account
+
+  -- Add to team
+  userTeam <- lmap CreateUserSparRegistrationError $ addUserToTeamSSO account tid (SSOIdentity ident Nothing Nothing)
+
+  -- Set up feature flags
+  let uid = userId (accountUser account)
+  lift $ initAccountFeatureConfig uid
+
+  -- Set handle
+  updateHandle' uid handle'
+
+  pure $! CreateUserResult account Nothing Nothing (Just userTeam)
+  where
+    lmap f =
```

Comment:
There's [mapLeft](https://hackage.haskell.org/package/either-5.0.2/docs/Data-Either-Combinators.html#v:mapLeft), or alternatively you could use [first](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Bifunctor.html#v:first).